models/aeif_psc_delta.cpp

```diff
@@ -0,0 +1,536 @@
+  const State_& s )
+{
+  assert( this != &s ); // would be bad logical error in program
+
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+    y_[ i ] = s.y_[ i ];
+  r_ = s.r_;
+  return *this;
+}
+
+/* ----------------------------------------------------------------
+ * Paramater and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::aeif_psc_delta::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::C_m, C_m );
+  def< double >( d, names::V_th, V_th );
+  def< double >( d, names::t_ref, t_ref_ );
+  def< double >( d, names::g_L, g_L );
+  def< double >( d, names::E_L, E_L );
+  def< double >( d, names::V_reset, V_reset_ );
+  def< double >( d, names::a, a );
+  def< double >( d, names::b, b );
+  def< double >( d, names::Delta_T, Delta_T );
+  def< double >( d, names::tau_w, tau_w );
+  def< double >( d, names::I_e, I_e );
+  def< double >( d, names::V_peak, V_peak_ );
+  def< double >( d, names::gsl_error_tol, gsl_error_tol );
+}
+
+void
+nest::aeif_psc_delta::Parameters_::set( const DictionaryDatum& d )
+{
+  updateValue< double >( d, names::V_th, V_th );
+  updateValue< double >( d, names::V_peak, V_peak_ );
+  updateValue< double >( d, names::t_ref, t_ref_ );
+  updateValue< double >( d, names::E_L, E_L );
+  updateValue< double >( d, names::V_reset, V_reset_ );
+
+  updateValue< double >( d, names::C_m, C_m );
+  updateValue< double >( d, names::g_L, g_L );
+
+  updateValue< double >( d, names::a, a );
+  updateValue< double >( d, names::b, b );
+  updateValue< double >( d, names::Delta_T, Delta_T );
+  updateValue< double >( d, names::tau_w, tau_w );
+
+  updateValue< double >( d, names::I_e, I_e );
+
+  updateValue< double >( d, names::gsl_error_tol, gsl_error_tol );
+
+  if ( V_reset_ >= V_peak_ )
+  {
+    throw BadProperty( "Ensure that V_reset < V_peak ." );
+  }
+
+  if ( Delta_T < 0. )
+  {
+    throw BadProperty( "Delta_T must be positive." );
+  }
+  else if ( Delta_T > 0. )
+  {
+    // check for possible numerical overflow with the exponential divergence at
+    // spike time, keep a 1e20 margin for the subsequent calculations
+    const double max_delta_arg =
+      std::log( std::numeric_limits< double >::max() / 1e20 );
+    if ( ( V_peak_ - V_th ) / Delta_T >= max_delta_arg )
+    {
+      throw BadProperty(
+        "The current combination of V_peak, V_th and Delta_T"
+        "will lead to numerical overflow at spike time; try"
+        "for instance to increase Delta_T or to reduce V_peak"
+        "to avoid this problem." );
+    }
+  }
+
+  if ( V_peak_ < V_th )
+  {
+    throw BadProperty( "V_peak >= V_th required." );
+  }
+
+  if ( C_m <= 0 )
+  {
+    throw BadProperty( "Ensure that C_m > 0" );
+  }
+
+  if ( t_ref_ < 0 )
+  {
+    throw BadProperty( "Ensure that t_ref >= 0" );
+  }
+
+  if ( tau_w <= 0 )
+  {
+    throw BadProperty( "All time constants must be strictly positive." );
```

Comment:
Since ``tau_w`` is now the only time constant, saying specifically that ``"tau_w must be strictly positive."`` would probably be better